youths/schema.py

```diff
@@ -128,20 +122,11 @@ def mutate(self, info, **kwargs):
         )
 
         if info.context.user.is_superuser:
-            try:
-                profile = Profile.objects.get(pk=profile_id)
-            except Profile.DoesNotExist:
-                raise GraphQLError(_("Invalid profile id."))
+            profile = Profile.objects.get(pk=profile_id)
         else:
-            try:
-                profile = Profile.objects.get(user=info.context.user)
-            except Profile.DoesNotExist:
-                raise GraphQLError(_("No profile found, please create one!"))
+            profile = Profile.objects.get(user=info.context.user)
 
-        try:
-            youth_profile = YouthProfile.objects.get(profile=profile)
-        except YouthProfile.DoesNotExist:
-            raise GraphQLError(_("No youth profile found to update."))
+        youth_profile = YouthProfile.objects.get(profile=profile)
```

Comment:
Hömpelipöm, not sure if I like this letting `DoesNotExist`s to just bubble up here. Possible issues that come to mind:

* When we have an actual bug somewhere which raises `DoesNotExist`, it won't get sent to Sentry

* When a query or mutation involves multiple objects, like here profile and youth profile, we return the same `OBJECT_DOES_NOT_EXIST` so the client is unable to determine which object is actually missing.

Reply:
Actually bubbled up exceptions are captured to sentry ATM. And the model is included in the error message (generated by django) and also in the path field. I'm still not sure about which exceptions we should send to sentry and I think I'm going to write a ticket about that. Because I think most of the time we don't want to send `OBJECT_DOES_NOT_EXIST` to sentry since it's most like due to someone playing around with the API or concurrent use of API where someone deletes a record just before someone else tries to read it.

Reply:
Yeah they are captured ATM, my bad. But as you said, most `OBJECT_DOES_NOT_EXIST`s should not be sent to Sentry, so once that is fixed, we'll get the problem I was talking about :)

I think that whatever a user throws at the API, it should not generate a Sentry error. Only bugs should. This probably means that in our code whenever we have a situation where we know a `DoesNotExist` can be raised, we should handle it somehow to prevent it from ending up in Sentry. And then unexcepted `DoesNotExist`s (bugs) will end up in Sentry.

And yes, Django adds the model in the message, but as the client may need to behave differently based on which object is missing, I don't think it's nice that the client then needs to parse the message, which kind of defeats the whole purpose of these error codes :)

Reply:
This is still the MVP which has been on my desk for too long due to A, B, C ... X, Y & Z. I'd like to improve this in the future since it's absolutely certain that we can't come up with a perfect solution.

The original goal for this story had nothing to do with sentry but it came along during the work. We need to have good guidelines about what to throw at sentry before we start implementing the logic for it.

The only option to keep `DoesNotExist`s away from sentry is definitely not to try-catch them all over the place which just pollutes the code.

I think that instead of using generic `OBJECT_DOES_NOT_EXIST` we could use more accurate error code mappings such as:
```
{
   Profile.DoesNotExist: "PROFILE_DOES_NOT_EXIST_ERROR",
   YouthProfile.DoesNotExist: "YOUTH_PROFILE_DOES_NOT_EXIST_ERROR",
   ...
}
```

Reply:
Fair enough :) And that seems like a viable solution indeed 👍